src/state/multicall/updater.tsx

```diff
@@ -1,269 +1,13 @@
-        ])
-        return c0.concat(c1)
-      }
-    }
-    console.error('Failed to fetch chunk', error)
-    throw error
-  }
-}
-
-/**
- * From the current all listeners state, return each call key mapped to the
- * minimum number of blocks per fetch. This is how often each key must be fetched.
- * @param allListeners the all listeners state
- * @param chainId the current chain id
- */
-export function activeListeningKeys(
-  allListeners: AppState['multicall']['callListeners'],
-  chainId?: number
-): { [callKey: string]: number } {
-  if (!allListeners || !chainId) return {}
-  const listeners = allListeners[chainId]
-  if (!listeners) return {}
-
-  return Object.keys(listeners).reduce<{ [callKey: string]: number }>((memo, callKey) => {
-    const keyListeners = listeners[callKey]
-
-    memo[callKey] = Object.keys(keyListeners)
-      .filter((key) => {
-        const blocksPerFetch = parseInt(key)
-        if (blocksPerFetch <= 0) return false
-        return keyListeners[blocksPerFetch] > 0
-      })
-      .reduce((previousMin, current) => {
-        return Math.min(previousMin, parseInt(current))
-      }, Infinity)
-    return memo
-  }, {})
-}
-
-/**
- * Return the keys that need to be refetched
- * @param callResults current call result state
- * @param listeningKeys each call key mapped to how old the data can be in blocks
- * @param chainId the current chain id
- * @param latestBlockNumber the latest block number
- */
-export function outdatedListeningKeys(
-  callResults: AppState['multicall']['callResults'],
-  listeningKeys: { [callKey: string]: number },
-  chainId: number | undefined,
-  latestBlockNumber: number | undefined
-): string[] {
-  if (!chainId || !latestBlockNumber) return []
-  const results = callResults[chainId]
-  // no results at all, load everything
-  if (!results) return Object.keys(listeningKeys)
-
-  return Object.keys(listeningKeys).filter((callKey) => {
-    const blocksPerFetch = listeningKeys[callKey]
-
-    const data = callResults[chainId][callKey]
-    // no data, must fetch
-    if (!data) return true
-
-    const minDataBlockNumber = latestBlockNumber - (blocksPerFetch - 1)
-
-    // already fetching it for a recent enough block, don't refetch it
-    if (data.fetchingBlockNumber && data.fetchingBlockNumber >= minDataBlockNumber) return false
+import { multicall } from './instance'
 
-    // if data is older than minDataBlockNumber, fetch it
-    return !data.blockNumber || data.blockNumber < minDataBlockNumber
-  })
-}
-
-export default function Updater(): null {
-  const dispatch = useAppDispatch()
-  const state = useAppSelector((state) => state.multicall)
-  // wait for listeners to settle before triggering updates
-  const debouncedListeners = useDebounce(state.callListeners, 100)
+// Create Updater wrappers that pull needed info from store
+export default function Updater() {
+  const Updater = multicall.Updater
```

Comment:
don't think we need to extract the updater here? It's also a big confusing because the component is called `Updater`

Reply:
Will fix, thanks!